reconcile/utils/deadmanssnitch_api.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from typing import (
+    Any,
+    Optional,
+    Self,
+)
+
+import requests
+from pydantic import BaseModel
+
+BASE_URL = "https://api.deadmanssnitch.com/v1/snitches"
+REQUEST_TIMEOUT = 60
+
+
+class DeadManssnitchException(Exception):
+    pass
+
+
+class Snitch(BaseModel):
+    token: str
+    href: str
+    name: str
+    tags: list[str]
+    notes: str
+    status: str
+    check_in_url: str
+    interval: str
+    alert_type: str
+    alert_email: list[str]
+    vault_data: Optional[str]
+
+    def get_cluster_name(self) -> str:
+        return self.name.split(".")[1]
```

Comment:
can use a mapping between cluster names and their `prometheusUrl` field to avoid string splitting? probably worth adding a `cluster_name` field somehow?

Reply:
added mapping between prometheus and cluster here https://github.com/app-sre/qontract-reconcile/blob/ff5294cb3123414309323dcd2b3916a5c5da49dd/reconcile/deadmanssnitch.py#L157